contrib/epee/include/span.h

```diff
@@ -71,7 +76,9 @@ namespace epee
     constexpr span() noexcept : ptr(nullptr), len(0) {}
     constexpr span(std::nullptr_t) noexcept : span() {}
 
-    constexpr span(T* const src_ptr, const std::size_t count) noexcept
+    //! Prevent derived-to-base conversions; invalid in this context.
+    template<typename U, typename = typename std::enable_if<safe_conversion<U>()>::type>
```

Comment:
What is that second anonymous parameter for ? It does not seem used. Is it ?

Reply:
The second template parameter is anonymous and unused. If `enable_if` is false, then the expression doesn't compute a valid type and the constructor is removed from the potential call set. If the `enable_if` is true, `::type` is `void` making the second template argument `void`. The general technique is known as SFINAE. In this particular case, a failed `enable_if` will result in a compiler error for no valid constructor call.

This logic isn't strictly needed, but was added to catch a [subtle bug with C++ classes and pointers](https://wiki.sei.cmu.edu/confluence/display/cplusplus/CTR56-CPP.+Do+not+use+pointer+arithmetic+on+polymorphic+objects). I just spotted [`has_virtual_destructor`](https://en.cppreference.com/w/cpp/types/has_virtual_destructor), and will add this check to the patch - preventing "polymorphic" classes. This check will be less necessary after that static check, but still helpful.

Reply:
And I just spotted that the `has_virtual_destructor` trait has a constraint that isn't workable (complete type). Crud, wont have that addition.